src/library/tableitemdelegate.h

```diff
@@ -1,25 +1,28 @@
-#ifndef TABLEITEMDELEGATE_H
-#define TABLEITEMDELEGATE_H
+#pragma once
 
 #include <QStyledItemDelegate>
+#include <QTableView>
 
-class QTableView;
-class QStyleOptionViewItem;
 
 class TableItemDelegate : public QStyledItemDelegate {
     Q_OBJECT
   public:
     explicit TableItemDelegate(QTableView* pTableView);
-    virtual ~TableItemDelegate();
+    ~TableItemDelegate() override = default;
```

Comment:
This is redundant on one hand, but I remember that there was an issue with a clang compiler warning. Is this still the case? Do we have a number of these compiler warnings elsewhere? 
If yes, we should think about a mass fix and drop a note here: https://www.mixxx.org/wiki/doku.php/coding_guidelines

Reply:
This is not redundant! It verifies that the destructor of the base class is virtual.

Reply:
Yes right, but is this important here?
But nevermind, it is a question if code style. We just need to state somewhere if we want that line or not. 

Does this issue a compiler warning, btw?

Reply:
I don't know and don't really care. If this pattern is not desired then simply remove all lines that match the regex "~.*override = default;" in a separate commit.

Reply:
I have read a bit what others think. While I originally was the opinion that it is unnecessary noise, this post convinced me that it is actually sensible https://github.com/isocpp/CppCoreGuidelines/issues/721#issuecomment-246466603

I have added a sentence to our style guides: 
> This applies also for default destructors even if it looks noisy.